app/main.py

```diff
@@ -0,0 +1,104 @@
+"""this is the entrypoint file"""
+# standard library imports
+from typing import List, Union
+import functools
+import logging
+
+# external library imports
+from fastapi import FastAPI, Response, status
+from fastapi.middleware.cors import CORSMiddleware
+
+# internal imports
+from app.models import website_data
+from app.helpers import code_compiler, estimate_complexity
+import app.settings as settings
+from app.helpers import decorators
+
+app = FastAPI(debug=settings.debug_state)
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=['*'],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+# Define the filter
+class HealthyEndpointFilter(logging.Filter):
+    def filter(self, record: logging.LogRecord) -> bool:
+        return '/healthy' not in record.args
+
+# Add filter to the logger
+logging.getLogger("uvicorn.access").addFilter(HealthyEndpointFilter())
+
+
+@app.get('/')
+@decorators.catchall_exceptions
+def base_url():
+    # include a list of the available endpoints
+    endpoints = '`/healthy` `/get_active_languages` `/estimate_complexity`'
+    return f'available endpoints are: {endpoints}'
+
+
+@app.get('/healthy')
+@decorators.catchall_exceptions
+def health_check():
+    return True
+
+
+@functools.lru_cache(maxsize=5)
+@app.get('/get_active_languages', status_code=status.HTTP_200_OK)
+@decorators.catchall_exceptions
+def programming_languages(response: Response) -> Union[List, None]:
```

Comment:
Todo: update this method to send a discord notification when no language list is returned by the compiler